plugins/modules/insights_register.py

```diff
@@ -144,7 +144,10 @@ def run_module():
             result['message'] = 'New machine-id created - ' + insights_name + ' has been registered'
             module.exit_json(**result)
         else:
-            subprocess.call([insights_name, '--register'])
+            if display_name:
+                subprocess.call([insights_name, '--register', '--display-name=' + display_name])
+            else:
+                subprocess.call([insights_name, '--register'])
```

Comment:
can you please avoid the duplication by factorizing the command line arguments? something like:
```python
args = [...]
if something_else:
    args.append(...)/.extend(...)
subprocess.call(args)
```